### ISSUE TITLE
Bug: [no-inferrable-types] False negative for parameter properties

### ISSUE THREAD

**Issue by controversial** [typescript-eslint/typescript-eslint#7278]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.tsx&code=MYGwhgzhAECyCeAVAphALtA3gKGtYA9gHboBOArsGgaQBS57QBmBBAXNEeQLYBGypaAF5oAZgA0jKXgD0M6ANI1JkBQA8ADsirIAJgzwbyvEAEtg0XmFIcufAcLGToczgQWklpBgEosB6AA3AlNdZlZJK1IAbgYAX2wEoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDNFpoAQwBGSWkVKoMkftA7RI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A&tokens=false

### Repro Code

```typescript
class MyTest {
  constructor(
    foo: number = 3,         // error, as expected
    public bar: number = 3,  // no error
  ) {
    void foo, bar;
  }
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-inferrable-types": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true
  }
}
```

### Expected Result

An error from `@typescript-eslint/no-inferrable-types` should appear for *both* parameters (both parameters include a type declaration for a trivially inferrable type).

### Actual Result

An error from `@typescript-eslint/no-inferrable-types` appears for the standard parameter but not for the parameter written as a [parameter property](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties).

### Additional Info

This issue appears with the default rule settings, i.e.
```ts
{
  ignoreParameters: false,
  ignoreProperties: false
}
```
This behavior might be considered expected when ignoreParameters and/or ignoreProperties is set to true (although how that behavior would be defined is a little murky, given that parameter properties are parameters *and* properties).

**Comment by bradzacher** [comment]
This is just a gap in the rule!

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/no-inferrable-types.ts#L254-L259
